### DISCUSSION TITLE
275. Facebook Playable Ads 与 H5 Game Engine 支持 调研

### DISCUSSION THREAD

**Discussion by ly525** [liuyanshi/gesang-fe-something#618] [小程序｜游戏｜跨端跨栈]
https://www.facebook.com/business/ads/playable-ad-format

# 简介
1. 通过简单的视频直观地了解下 Facebook Playable Ads 
   1. [Facebook Playable Ad](https://www.youtube.com/watch?v=jb-o1vrDZ9E)

# Facebook Playable Ads 规范
1. [Specifications for Playable Ads
](https://www.facebook.com/business/help/412951382532338?helpref=faq_content)

### 规范限制
1. The primary restriction is that the export must be a single file, with everything inlined in to index.html。该游戏相关的所有资源必须放在 html 中，不允许发起请求

# 引擎调研

### 引擎不支导出 playable ad 
1. GDevelop 目前不支持，计划支持：https://trello.com/c/thiOmEM2/232-add-export-for-fb-playable-ads
2. 国内各大引擎目前均不支持：egret

### 引擎支持导出 Playable Ad
1. [CONSTRUCT 3 R134
](https://www.construct.net/en/make-games/releases/beta/r134)

支持导出 Facebook Playable Ad， 但是该功能为付费版

> ![image](https://user-images.githubusercontent.com/12668546/55156785-c6df2300-5195-11e9-9878-2202a0f5c54a.png)
> 相关讨论：https://www.construct.net/en/forum/construct-3/how-do-i-8/facebook-playable-ads-140842

2. [2dkit](http://2dkit.com/)
> CREATE GAMES ONCE. PLAY ANYWHERE.
A professional-grade game engine and tools for native-quality Messenger Games and Playable Ads that load instantly.

进展：注册成功，但始终无法登录。多次尝试，均失败。请其他同事注册过，均失败。已经尝试在 Twitter 上和其联系

3. [Adobe Animate CSS discussion](https://forums.adobe.com/thread/2597639) 
进展： 需要尝试

4. https://www.synative.com/index
进展：已经注册账号，等待审核通过

### 编译各大引擎导出的项目
1. 编译 CocosCreator 导出的项目：https://github.com/chongshengzhujue/playableFBCompile
进展：demo 可以跑通，但需要结合 CocosCreator 进行具体实验

### Playable Ad 相关资讯
1. [MarketJS CEO 谈 Playable Ads](https://soundcloud.com/marketjs/playable-ads-b2b-games-podcast-episode-2)

**Comment by NicholasPapa** [comment]
2dkit注册不是立即生效的，好像是要审核，亲测隔天会发注册成功的邮件，并且可以登录。
